### ISSUE TITLE
Reading Stata value labels fails for all value labels prior to an empty value label

### ISSUE THREAD

**Issue by NilsEnevoldsen** [WizardMac/ReadStat#219]
Reported as https://github.com/tidyverse/haven/issues/551

Variables whose value labels appear on the label list prior to one with an empty value label are not assigned their value labels.

Step 1:

```stata
input x y z
0 0 0
end

lab def xlab 0 "Foo"
lab val x xlab

lab def ylab 0 "Bar"
lab val y ylab

lab def ylab 0 "", modify // This is one way to create an "empty" value label. It does *not* create a value label with an empty string for value 0.

lab def zlab 0 "FooBar"
lab val z zlab

label list
save example_input
```

Step 2: `readstat example_input.dta example_output.dta`

Actual `example_output.dta `:

![Screen Shot 2020-11-22 at 9 06 35 PM](https://user-images.githubusercontent.com/222743/99924047-ae2a6600-2d06-11eb-9bd9-990c3907bcfa.png)

Expected `example_output.dta `:

![Screen Shot 2020-11-22 at 9 06 49 PM](https://user-images.githubusercontent.com/222743/99924046-ac60a280-2d06-11eb-9c6c-ad85bc6b3906.png)

(Minor detail: the blueness of the `0` indicates the presence of an [empty] value label for variable `y`; I have no strong opinion on whether this value label should be preserved in the output file.)

**Comment by evanmiller** [comment]
Can you please provide a copy of `example_input.dta`? Thanks

**Comment by NilsEnevoldsen** [comment]
Yes, I have attached four `example_input`s. I used Stata 14 to generate them with the following code:

```stata
save example_input_14
saveold example_input_13, version(13)
saveold example_input_12, version(12)
saveold example_input_11, version(11)
```

`example_input_14.dta` (version 118) and `example_input_13.dta` (version 117) exhibit the incorrect behavior when processed by ReadStat. `example_input_12.dta` (version 115) and `example_input_11.dta` (also version 115? I'm unclear on this) exhibit correct behavior when processed by ReadStat. (In the latter cases, they do both drop ylab, which I don't feel strongly about.)

[example_input.zip](https://github.com/WizardMac/ReadStat/files/5610872/example_input.zip)

**Comment by evanmiller** [comment]
Thanks. See if this commit fixes things:

https://github.com/WizardMac/ReadStat/commit/263ba39e4f29dc50f071e83c50c326f42c9207ac

**Comment by NilsEnevoldsen** [comment]
I get an error on every DTA file with 1.1.5-rc0.

```shell
% readstat example_input_16.dta example_output_16.dta
Error beginning row: Unable to write data
Converted 1 variables and 1 rows in 0.00 seconds
Error processing example_input_16.dta: The parsing was aborted (callback returned non-zero value)
```

**Comment by evanmiller** [comment]
Please try the `dev` branch rather than the RC

**Comment by NilsEnevoldsen** [comment]
Ah, right, I just saw 93e33e25b21d8a0894b73a776cc99529e0bf4986

**Comment by NilsEnevoldsen** [comment]
Yes, this fix works for me, with no immediately apparent side effects. I tested example DTAs from Stata versions 11 through 16, as well as my own "in the wild" problematic example, and that reported at tidyverse/haven#551.

Thank you for your prompt fix! Will this make it into 1.1.5?

**Comment by evanmiller** [comment]
> Will this make it into 1.1.5?

Yes
